src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvOutputConsumer.cs

```diff
@@ -81,7 +83,7 @@ public class LibuvOutputConsumer
                 }
                 finally
                 {
-                    _pipe.Advance(result.Buffer.End);
+                    _pipe.Advance(consumed);
```

Comment:
When error happens during execution of a write request abort gets _posted_ and if `Advance` relieves back pressure there is a race happening between flush completion and cancellation token callback. So not releasing back pressure in case of an error prevents the race.